app/models/katello/concerns/host_managed_extensions.rb

```diff
@@ -61,6 +63,21 @@ def update_os_from_facts
         scoped_search :relation => :host_traces, :on => :application, :complete_value => true, :rename => :trace_app, :only_explicit => true
         scoped_search :relation => :host_traces, :on => :app_type, :complete_value => true, :rename => :trace_app_type, :only_explicit => true
         scoped_search :relation => :host_traces, :on => :helper, :complete_value => true, :rename => :trace_helper, :only_explicit => true
+
+        scoped_search relation: :pools, on: :pools_expiring_in_days, ext_method: :find_with_expiring_pools, only_explicit: true
+
+        def self.find_with_expiring_pools(_key, _operator, days_from_now)
```

Comment:
Does this simplification work?

```
        def self.find_with_expiring_pools(_key, _operator, days_from_now)                                                                                     
          host_ids = with_pools_expiring_in_days(days_from_now).ids                                                                                           
          condition = if host_ids.any?                                                                                                                        
                        "hosts.id IN (#{host_ids.join(', ')})"                                                                                                
                      else                                                                                                                                    
                        '1=0'                                                                                                                                 
                      end                                                                                                                                     
          { conditions: condition }                                                                                                                           
        end 
```

I don't understand what the initial definition of `scoped_search_params` is needed for (the case when there are no matching hosts). It seems like if there are no matching hosts then we weren't going to show any results anyway - regardless of any addition scoped search filters, so no need to join pools, filter them, etc.

Reply:
The `:conditions` value becomes the `WHERE` clause.  I agree that it could use some simplification, but I don't think `WHERE 1=0` would work..?
(see [this](https://github.com/Katello/katello/pull/8606#discussion_r395337525) comment for why we need the initial definition.)

Reply:
How about something like `hosts.id = 0` ?

Reply:
But 1=0 is three characters long ;)

Reply:
Also, please add a test for this in HostManagedExtensionsTest by calling search_for